### DISCUSSION TITLE
Merge of minio main branch into gateway-1.1 branch

### DISCUSSION THREAD

**Discussion by Tharyrok** [juicedata/juicefs#4435] [General]
Hi everyone,
First, thank you very much for creating and evolving JuiceFS.

Would it be possible to release a new version of minio that contains the change from the main branch?

I am particularly interested in this PR https://github.com/juicedata/minio/pull/47.

Currently, I have found a workaround by enabling an etcd cluster in the environment variables, which allows me to have multiple buckets and users. I attach policies to groups, but it would be much easier for me to do this directly on the bucket. Also, lifecycle management would be great :)

I wanted to clone the main branch and compile this version, but I noticed that there are changes in gateway-1.1 that are not in main.

So my questions are:
- Would it be possible to have an ETA on a new release in the gateway-1.1 branch?
- If this is not a priority, how can I build a new version of minio that includes the change from the main branch?

Thanks in advance for your answers.

**Comment by zhijian-pro** [comment]
Try the branch https://github.com/juicedata/minio/commits/refactor, which still has some problems, so it's not merged
In addition, our main branch of juicefs has already implemented similar functionality support, such as multi-user, bucket policies, and so on. You can try it too.
